src/odemis/driver/xt_client.py

```diff
@@ -164,6 +324,34 @@ def __init__(self, name, role, children, address, daemon=None,
                 self._detector = Detector(parent=self, daemon=daemon, **ckwargs)
             self.children.value.add(self._detector)
 
+    def get_value_from_software_version(self, element: str, info: str) -> str:
+        """
+        Get an elements's value from xtadapter's software version.
+
+        :param element: The element whose value needs to be extracted.
+        :param info: The xtadapter's software version information.
+        :return: The element's value, if not found return None.
+
+        """
+        # Example information: xt: 1.1.1; xtadapter: 1.2.1
+        items = info.split(";")
+        for item in items:
+            if element in item.lower():
+                value = item.split(":")[1].strip()
+                return value
+        return None
```

Comment:
So far all the functions on this class correspond exactly to a function on the xtadapter. I'm not sure if this function actually fits here. Also I think you can just replace this by a single regex, so then you might as well just put that on line 243 for instance for the xtadapter element
```
current_version = re.search("xtadapter: (\s*([\d.]+))", info)
current_version = current_version.group(1) if current_version is not None else None
```